qiskit/extensions/quantum_initializer/initializer.py

```diff
@@ -238,6 +238,8 @@ def initialize(self, params, qubits):
     # TODO: avoid explicit reset if compiler determines a |0> state
     if isinstance(qubits, QuantumRegister):
         qubits = qubits[:]
+    else:
+        qubits = _convert_to_bits([qubits], [qbit for qreg in self.qregs for qbit in qreg])[0]
```

Comment:
because, as `barrier`, `initialize` is also very much non-standard. `_convert_to_bits` expects a list and `circuit.initialize(init_vector, slice(1, 3))` will pass a slice to the function. I could move the recursion one level up.... but I'm pretty sure that will break something else :S